datastream-server/src/test/java/com/linkedin/datastream/server/zk/TestZkAdapter.java

```diff
@@ -758,32 +765,37 @@ public void testDeleteTasksWithPrefix() {
     Mockito.verify(zkClient, Mockito.never()).getChildren(any(), anyBoolean());
 
     List<String> leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(false);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(true);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     updateInstanceAssignment(adapter, adapter.getInstanceName(), Collections.emptyList());
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(false);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 2);
+    Assert.assertEquals(leftOverTasks.size(), 3);
 
     adapter.cleanUpOrphanConnectorTasks(true);
 
     leftOverTasks = zkClient.getChildren(KeyBuilder.connector(testCluster, connectorType));
-    Assert.assertEquals(leftOverTasks.size(), 0);
+    Assert.assertEquals(leftOverTasks.size(), 1);
```

Comment:
Should we also assert that one znode left is actually `lock`?

Reply:
done.